hazelcast-client/src/test/java/com/hazelcast/client/cache/nearcache/ClientNearCachePreloaderTest.java

```diff
@@ -0,0 +1,258 @@
+    }
+
+    @SuppressWarnings("SameParameterValue")
+    private void preloadNearCache(File preloaderFile, int keyCount, boolean useStringKey) {
+        CacheSimpleConfig cacheSimpleConfig = createCacheSimpleConfig(inMemoryFormat, keyCount);
+        serverInstance.getConfig().addCacheConfig(cacheSimpleConfig);
+
+        // populate the member side cache, so we have the values to populate the Near Cache
+        ICache<Object, String> memberCache = serverInstance.getCacheManager().getCache(DEFAULT_CACHE_NAME);
+        for (int i = 0; i < keyCount; i++) {
+            Object key = useStringKey ? "key-" + i : i;
+            memberCache.put(key, "value-" + i);
+        }
+
+        // start the client which will kick off the Near Cache pre-loader
+        NearCacheConfig nearCacheConfig = getNearCacheConfig(keyCount, preloaderFile.getAbsolutePath());
+        NearCacheTestContext context = createNearCacheTest(DEFAULT_CACHE_NAME, nearCacheConfig);
+        assertNearCacheSizeEventually(context, keyCount);
+    }
+
+    private NearCacheConfig getNearCacheConfig(int size, String preloaderFileName) {
+        NearCacheConfig nearCacheConfig = createNearCacheConfig(inMemoryFormat)
+                .setCacheLocalEntries(true)
+                .setInvalidateOnChange(false);
+
+        configureEvictionConfig(nearCacheConfig.getEvictionConfig(), size);
+
+        nearCacheConfig.getPreloaderConfig()
+                .setEnabled(true)
+                .setStorageFileName(preloaderFileName);
+
+        return nearCacheConfig;
+    }
+
+    protected CacheSimpleConfig createCacheSimpleConfig(InMemoryFormat inMemoryFormat, int size) {
+        CacheSimpleConfig cacheSimpleConfig = new CacheSimpleConfig()
+                .setName(DEFAULT_CACHE_NAME)
+                .setInMemoryFormat(inMemoryFormat);
+
+        configureEvictionConfig(cacheSimpleConfig.getEvictionConfig(), size);
+
+        return cacheSimpleConfig;
+    }
+
+    private void configureEvictionConfig(EvictionConfig evictionConfig, int size) {
+        evictionConfig
+                .setMaximumSizePolicy(MaxSizePolicy.ENTRY_COUNT)
+                .setSize(size)
+                .setEvictionPolicy(EvictionPolicy.LRU);
+    }
+
+    private static void populateNearCache(NearCacheTestContext context, int keyCount, boolean useStringKey) {
+        for (int i = 0; i < keyCount; i++) {
+            Object key = useStringKey ? "key-" + i : i;
+            context.cache.put(key, "value-" + i);
+            context.cache.get(key);
+        }
+    }
+
+    private static long getPersistenceCount(NearCacheTestContext context) {
+        return context.nearCache.getNearCacheStats().getPersistenceCount();
+    }
+
+    private static void waitForNearCachePersistence(NearCacheTestContext context, int persistenceCount) {
+        long oldPersistenceCount = getPersistenceCount(context);
+        while (getPersistenceCount(context) <= oldPersistenceCount + persistenceCount) {
```

Comment:
I changed this and also transformed the tests to use the new unified Near Cache test approach with the new context. I was actually missing a test for the `IMap` scenario, just the `JCache` path was tested. With the unified approach this was easy to add.